interfaces/thirdparty/cp2k.py

```diff
@@ -499,20 +499,20 @@ def _parsemol(self):
         # make lines shorter
         inp = self.settings.input.force_eval.subsys
         # add cell information
-        nDim = len(self.molecule.lattice)
+        dim = len(self.molecule.lattice)
         keys = ['A', 'B', 'C']
         periodic = ['X', 'XY', 'XYZ']
-        for iDim in range(0, nDim):
-            inp.cell[keys[iDim]] = "{:} {:} {:}".format(
-                *self.molecule.lattice[iDim])
-        if nDim > 0:
-            inp.cell.periodic = periodic[nDim-1]
+        for i in range(dim):
+            inp.cell[keys[i]] = "{:.6f} {:.6f} {:.6f}".format(
+                *self.molecule.lattice[i])
+        if dim > 0:
+            inp.cell.periodic = f"{periodic[dim - 1]:.6f}"
```

Comment:
```suggestion
            inp.cell.periodic = periodic[dim - 1]
```
`periodic` is a lists of strings, not floats.